### PR TITLE
lint: forbid whitespace in empty list decls

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR fixes cases where empty list declarations had whitespace in their parentheses, and adds a check_grep lint to forbid it.
## Why It's Good For The Game
Shit's dumb
## Testing
Visual inspection
<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
<hr>

## Changelog
NPFC

### CODE REVIEW

**Review by AffectedArc07 — APPROVED** [review]

File: tools/ci/check_grep2.py [hunk 1]
```diff
@@ -151,6 +151,11 @@ def check_href_styles(idx, line):
     if HREF_OLD_STYLE.search(line):
         return [(idx + 1, "BYOND requires internal href links to begin with \"byond://\"")]
 
+EMPTY_LIST_WHITESPACE = re.compile(r"list\(\s+\)")
```
[the remark this answers is not in the document]
Comment (lewcc) [reply tools/ci/check_grep2.py:154]: I don't think this suggested change is correct, losing the quote (and not removing the escaping backslash)

[the remark this answers is not in the document]
Comment (Contrabang) [reply tools/ci/check_grep2.py:154]: oops, fixed

File: tools/ci/check_grep2.py [hunk 2]
```diff
@@ -151,6 +151,11 @@ def check_href_styles(idx, line):
     if HREF_OLD_STYLE.search(line):
         return [(idx + 1, "BYOND requires internal href links to begin with \"byond://\"")]
 
+EMPTY_LIST_WHITESPACE = re.compile(r"list\(\s+")
```
Comment (Contrabang) [inline tools/ci/check_grep2.py:154]: It seems that whitespace counts newlines, whoops
```suggestion
EMPTY_LIST_WHITESPACE = re.compile(r"list\(\s+/)")
```

Comment (warriorstar-orion) [reply tools/ci/check_grep2.py]: Made the regex catch nearly issue: https://regex101.com/r/WCCmUf/1

PTAL

File: tools/ci/check_grep2.py [hunk 3]
```diff
@@ -151,7 +151,10 @@ def check_href_styles(idx, line):
     if HREF_OLD_STYLE.search(line):
         return [(idx + 1, "BYOND requires internal href links to begin with \"byond://\"")]
 
-EMPTY_LIST_WHITESPACE = re.compile(r"list\(\s+")
+# TODO: This finds most cases except for e.g. `list(1, 2, 3 )`
+# Find a way to include this without breaking macro/tab-aligned versions such as `list(		\`
+# Maybe even make sure it doesn't include comments, idk
```
Comment (lewcc) [inline tools/ci/check_grep2.py:156]: Planning on handling this TODO in the PR?

Comment (warriorstar-orion) [reply tools/ci/check_grep2.py:156]: I'm leaving it for better minds than me.

**Review by Contrabang — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Pull request merged**